File: easy/1_two_number_sum/two_number_sum.java
import java.util.*;

class Program{
    public static void main(String[] args) {
        int[] Array = {3, 5, -4, 8, 11, 1, -1, 6};
        int targetSum = 10;
        int [] Result = twoNumberSum(Array,targetSum);
        System.out.println(Arrays.toString(Result)); 
    }
	public static int[] twoNumberSum(int[] array, int targetSum){
		Set<Integer> hashTable = new HashSet<>();
		for(int num:array){
			int remainder = targetSum - num;
			if (hashTable.contains(remainder)){
				return new int[]{remainder,num};
			}
			else{
				hashTable.add(num);
			}
		}
		return new int[0];
	}
}

File: easy/1_two_number_sum/two_number_sum.py
# TS O(n)
def twoNumberSum(array, targetSum):
	hash_table = {} # hash table 
	for num in array: 
		remainder = targetSum - num 
		if remainder in hash_table:
			return [remainder,num]
		else: 
			hash_table[num]=True
	return []



if __name__ == "__main__":
    array = [3, 5, -4, 8, 11, 1, -1, 6]
    targetSum =10
    result = twoNumberSum(array, targetSum)
    print(result)